### PR TITLE
Cloud Free Trial Button Enable/Disable Option

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
The following command need to run
    php artisan migrate

### CODE REVIEW

**Review by ajaz-ul-haq — APPROVED** [review]

File: app/Http/Controllers/Tenancy/TenantController.php [hunk 1]
```diff
@@ -37,9 +38,22 @@ public function viewTenant()
         $response = json_decode($responseBody, true);
 
         $de = collect($response['message'])->paginate(5);
+        $cloudButton = StatusSetting::where('id', '1')->value('cloud_button');
```
Comment (laksh-22) [inline app/Http/Controllers/Tenancy/TenantController.php:41]: you can directly use StatusSetting::value('cloud_button');

File: app/Http/Controllers/Tenancy/TenantController.php [hunk 2]
```diff
@@ -37,9 +38,22 @@ public function viewTenant()
...
 
-        return view('themes.default1.tenant.index', compact('cloud', 'de'));
+        return view('themes.default1.tenant.index', compact('cloud', 'de', 'cloudButton'));
+    }
+
+    public function enableCloud(Request $request)
+    {
+        try {
+            $debug = $request->input('debug');
+            $debug == 'true' ? StatusSetting::where('id', '1')->update(['cloud_button' => '1']) : StatusSetting::where('id', '1')->update(['cloud_button' => '0']);
```
Comment (laksh-22) [inline app/Http/Controllers/Tenancy/TenantController.php:51]: you can directly use $request->input('debug'); instead of creating a variable

Comment (laksh-22) [reply app/Http/Controllers/Tenancy/TenantController.php:51]: and StatusSetting::first()->update(['cloud_button' => '1']) might also work

**Review by laksh-22 — DISMISSED** [review]

File: app/Http/Controllers/Tenancy/TenantController.php [hunk 3]
```diff
@@ -37,9 +38,20 @@ public function viewTenant()
...
 
-        return view('themes.default1.tenant.index', compact('cloud', 'de'));
+        return view('themes.default1.tenant.index', compact('cloud', 'de', 'cloudButton'));
+    }
+
+    public function enableCloud(Request $request)
+    {
+        try {
+            $request->input('debug') == 'true' ? StatusSetting::where('id', '1')->update(['cloud_button' => '1']) : StatusSetting::where('id', '1')->update(['cloud_button' => '0']);
+
```
Comment (ajaz-ul-haq) [inline app/Http/Controllers/Tenancy/TenantController.php:50]: **StatusSetting::first()->update(['cloud_button' => (int) $request->input('debug')]);**

Above can be replaced with this, i think Can you check please

**Review by ajaz-ul-haq — DISMISSED** [review]

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Last commit
image
image
**Unable to test getting below error**
image

**Comment by SarfrazS** [comment]
Please make it first later capital.
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Please don't miss basic things.

**Review by SarfrazS — DISMISSED** [review]
Pointed issues were fixed.
**When setting is enabled then faveo Cloud button is showing on the client panel. (for logged-in users)**
image
When the setting is Disabled then faveo Cloud button is not showing on the client panel. (for logged-in users)
image
image

Query:- Cloud Trail option is enabled and the user logged in user can see faveo cloud after that admin is set as Disabled, once the user refreshes the URL then only it is vanishing. Is it fine?

Note:- if Cloud trail set as disabled then i need to check cloud order is showing or not on the final build with live cloud data. here i am not able to test it.

**Pull request merged**